simulation/modules/plot.py

```diff
@@ -11,61 +13,85 @@
 @attr.s(kw_only=True)
 class PlotState(ModuleState):
     step_num: int = attr.ib(default=0)
+    time_steps: np.ndarray = attr.ib(factory=lambda: np.zeros(MAX_ARRAY_LENGTH, dtype=float))
+
+    # arrays to plot against time_steps
     fungal_burdens: np.ndarray = attr.ib(factory=lambda: np.zeros(MAX_ARRAY_LENGTH, dtype=int))
     macrophage_counts: np.ndarray = attr.ib(factory=lambda: np.zeros(MAX_ARRAY_LENGTH, dtype=int))
     neutrophil_counts: np.ndarray = attr.ib(factory=lambda: np.zeros(MAX_ARRAY_LENGTH, dtype=int))
-    time_steps: np.ndarray = attr.ib(factory=lambda: np.zeros(MAX_ARRAY_LENGTH, dtype=float))
 
 
 class Plot(Module):
     name = 'plot'
+    defaults = {'plot_output_folder': 'output/plots/'}
     StateClass = PlotState
 
+    def initialize(self, state: State):
+        self.advance(state, state.time)
+
+        return state
+
     def advance(self, state: State, previous_time: float):
-        step_num = state.plot.step_num
+        plot: PlotState = state.plot
+        step_num = plot.step_num
 
-        state.plot.time_steps[step_num] = state.time
-        state.plot.fungal_burdens[step_num] = len(state.fungus.cells.alive())
-        state.plot.macrophage_counts[step_num] = len(state.macrophage.cells.alive())
-        state.plot.neutrophil_counts[step_num] = len(state.neutrophil.cells.alive())
-        # define additional values to be plotted here
+        plot.time_steps[step_num] = state.time
+        plot.fungal_burdens[step_num] = len(state.fungus.cells.alive())
+        plot.macrophage_counts[step_num] = len(state.macrophage.cells.alive())
+        plot.neutrophil_counts[step_num] = len(state.neutrophil.cells.alive())
 
-        state.plot.step_num += 1
+        plot.step_num += 1
```

Comment:
I'm glad you recognized the tricky fact here that you *had* to increment `plot.step_num`, and not `step_num`. FWIW, you *could* have changed either `state.plot.step_num` or `plot.step_num` and had the correct effect though. If you are unclear as to why any of this is the case, feel free to talk to me.